src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,148 @@
+import { type ChangeEvent, useCallback, useEffect, useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { LifeCycleStatus, SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+function hasMaxSlippage(
+  statusData: any,
+): statusData is { maxSlippage: number } {
+  return statusData?.maxSlippage !== undefined;
+}
+
+function updateMaxSlippage(
+  status: LifeCycleStatus,
+  newMaxSlippage: number,
+): LifeCycleStatus {
+  if (['init', 'amountChange'].includes(status.statusName)) {
```

Comment:
I think we can remove init from this file, slippage should only update amountChange lifecycle